vulture_os/toolkit/api_parser/catonetworks/catonetworks.py

```diff
@@ -0,0 +1,156 @@
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+
+__author__ = "Gaultier Parain"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'Catonetworks API Parser toolkit'
+__parser__ = 'CATONETWORKS'
+
+import time
+import json
+import logging
+import requests
+
+from django.conf import settings
+from toolkit.api_parser.api_parser import ApiParser
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class CatonetworksAPIError(Exception):
+    pass
+
+
+class CatonetworksParser(ApiParser):
+
+    URL = "https://api.catonetworks.com/api/v1/graphql2"
+
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.catonetworks_api_key = data["catonetworks_api_key"]
+        self.catonetworks_account_id = data["catonetworks_account_id"]
+
+        self.marker = data.get("catonetworks_marker", "")
+        self.event_filter_string = ""
+        self.event_subfilter_string = ""
+
+    def execute_query(self, query):
+        retry_count = 0
+        data = {'query':query}
+        headers = { 'x-api-key': self.catonetworks_api_key,'Content-Type':'application/json'}
+        while True:
+            if retry_count > 5:
+                logger.error(f"[{__parser__}]:execute_query: FATAL ERROR retry count exceeded", extra={'frontend': str(self.frontend)})
+                raise CatonetworksAPIError("FATAL ERROR retry count exceeded")
+            try:
+                response = requests.post(
+                    url=self.URL,
+                    data=json.dumps(data).encode("ascii"),
+                    headers=headers,
+                    proxies=self.proxies,
+                    verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl,
+                    timeout=30
+                )
+            except Exception as e:
+                logger.info(f"[{__parser__}]:execute_query: {e}", extra={'frontend': str(self.frontend)})
+                time.sleep(20)
+                self.update_lock()
+                retry_count += 1
+                continue
+            break
+        response.raise_for_status()
+        result = response.json()
+        if "errors" in result:
+            logger.error(f"[{__parser__}]:execute_query: Error while getting logs. reason : {response.content}", extra={'frontend': str(self.frontend)})
+            raise CatonetworksAPIError(f"Error while getting logs. reason : {response.content}")
```

Comment:
Not gonna discuss the cases where you get to that point without having the `response.raise_for_status()` actually raise an exception, but why use `response.content` when you check for what seems to be a field presenting the errors to the user?
Why not use `result['errors']` in logs instead?